ledger/applications_test.go

```diff
@@ -1465,3 +1467,393 @@ return
+		a.Equal(basics.StateSchema{NumByteSlice: 3}, app().LocalStateSchema)
+		a.Zero(app().GlobalState)
+		a.Equal(proto.MinBalance+
+			proto.AppFlatParamsMinBalance+
+			2*(proto.SchemaMinBalancePerEntry+proto.SchemaUintMinBalance),
+			mbr(addrs[0]))
+		a.Equal(proto.MinBalance, mbr(addrs[1]))
+
+		// call with two args, fails from lack of globals
+		dl.txn(&txntest.Txn{
+			Type:            protocol.ApplicationCallTx,
+			Sender:          addrs[0],
+			ApplicationID:   appID,
+			ApplicationArgs: [][]byte{{'A'}, {'B'}},
+		}, "store bytes count 1 exceeds schema bytes count 0")
+
+		update := txntest.Txn{
+			Type:              protocol.ApplicationCallTx,
+			Sender:            addrs[1], // Use another sender, to follow MBR changes
+			ApplicationID:     appID,
+			OnCompletion:      transactions.UpdateApplicationOC,
+			ApprovalProgram:   app().ApprovalProgram, // keep it
+			GlobalStateSchema: basics.StateSchema{NumByteSlice: 1},
+		}
+		// update the global schema to allow 1 byteslice
+		if ver >= 42 {
+			dl.txn(&update)
+		} else {
+			dl.txn(&update, "inappropriate non-zero tx.GlobalStateSchema")
+			return // no more tests, growing is disallowed
+		}
+		a.EqualValues(1, app().GlobalStateSchema.NumByteSlice)
+		a.Zero(app().GlobalStateSchema.NumUint)                              // cleared
+		a.Equal(basics.StateSchema{NumByteSlice: 3}, app().LocalStateSchema) // unchanged
+		a.Empty(app().GlobalState)
+		a.Equal(proto.MinBalance+proto.AppFlatParamsMinBalance, mbr(addrs[0]))
+		a.Equal(proto.MinBalance+
+			1*(proto.SchemaMinBalancePerEntry+proto.SchemaBytesMinBalance)+
+			0*(proto.SchemaMinBalancePerEntry+proto.SchemaUintMinBalance),
+			mbr(addrs[1]))
+
+		// call with two args can now suceed
+		dl.txn(&txntest.Txn{
+			Type:            protocol.ApplicationCallTx,
+			Sender:          addrs[0],
+			ApplicationID:   appID,
+			ApplicationArgs: [][]byte{{'A'}, {'B'}},
+		})
+		a.EqualValues(1, app().GlobalStateSchema.NumByteSlice)
+		a.Len(app().GlobalState, 1) // 'A' has been added
+
+		// same global ('A') can be changed
+		dl.txn(&txntest.Txn{
+			Type:            protocol.ApplicationCallTx,
+			Sender:          addrs[0],
+			ApplicationID:   appID,
+			ApplicationArgs: [][]byte{{'A'}, {'C'}},
+		})
+
+		// new global ('X') is too many
+		dl.txn(&txntest.Txn{
+			Type:            protocol.ApplicationCallTx,
+			Sender:          addrs[0],
+			ApplicationID:   appID,
+			ApplicationArgs: [][]byte{{'X'}, {'Y'}},
+		}, "store bytes count 2 exceeds schema bytes count 1")
+
+		update.GlobalStateSchema = basics.StateSchema{NumByteSlice: 2}
+		dl.txn(&update)
+		a.Equal(proto.MinBalance+proto.AppFlatParamsMinBalance, mbr(addrs[0]))
+		a.Equal(proto.MinBalance+
+			2*(proto.SchemaMinBalancePerEntry+proto.SchemaBytesMinBalance)+
+			0*(proto.SchemaMinBalancePerEntry+proto.SchemaUintMinBalance),
+			mbr(addrs[1]))
+		// ok now
+		dl.txn(&txntest.Txn{
+			Type:            protocol.ApplicationCallTx,
+			Sender:          addrs[0],
```

Comment:
Is the sender here supposed to be different than who provides the MBR? (I think it is and that is the point)

Reply:
Here it doesn't matter, I'm just trying to show that the the GlobalSchema has indeed been grown, because this call, which will allocate another global, succeeds.

(In the previous few lines, I am testing that the MBR has moved around properly.)